pancancer_utilities/tcga_utilities.py

```diff
@@ -0,0 +1,180 @@
+"""
+Functions for preprocessing TCGA expression data and mutation status labels.
+
+Most functions are adapted from:
+https://github.com/greenelab/BioBombe/blob/master/9.tcga-classify/scripts/tcga_util.py
+"""
+import os
+
+import pandas as pd
+from sklearn.preprocessing import StandardScaler
+
+def process_y_matrix(
+    y_mutation,
+    y_copy,
+    include_copy,
+    gene,
+    sample_freeze,
+    mutation_burden,
+    filter_count,
+    filter_prop,
+    output_directory,
+    hyper_filter=5,
+):
+    """
+    Combine copy number and mutation data and filter cancer-types to build y matrix
+
+    Arguments
+    ---------
+    y_mutation: Pandas DataFrame of mutation status
+    y_copy: Pandas DataFrame of copy number status
+    include_copy: boolean if the copy number data should be included in status calc
+    gene: string indicating gene of interest (used for writing proportion file)
+    sample_freeze: pandas dataframe storing which samples to use
+    mutation_burden: pandas dataframe storing log10 mutation counts
+    filter_count: the number of positives or negatives required per cancer-type
+    filter_prop: the proportion of positives or negatives required per cancer-type
+    output_directory: the name of the directory to store the gene summary
+    hyper_filter: the number of std dev above log10 mutation burden to filter
+
+    Returns
+    -------
+    Write file of cancer-type filtering to disk and output a processed y vector
+    """
+    if include_copy:
+        y_df = y_copy + y_mutation
+    else:
+        y_df = y_mutation
+
+    y_df.loc[y_df > 1] = 1
+    y_df = pd.DataFrame(y_df)
+    y_df.columns = ["status"]
+
+    y_df = (
+        y_df.merge(
+            sample_freeze, how="left", left_index=True, right_on="SAMPLE_BARCODE"
+        )
+        .set_index("SAMPLE_BARCODE")
+        .merge(mutation_burden, left_index=True, right_index=True)
+    )
+
+    # Get statistics per gene and disease
+    disease_counts_df = pd.DataFrame(y_df.groupby("DISEASE").sum()["status"])
```

Comment:
Ok based on this I think I might be confused as to the format of `y_df`

Reply:
This is another holdover from Greg's code, but I think the merge calls above add a column for the cancer type of each sample (called `DISEASE`), then the code below filters cancer types (diseases) based on the proportion of positively labeled samples. It doesn't make sense to do classification if a gene isn't mutated in any samples (or is mutated in almost every sample), so we just ignore those cancer types. 

Let me know if you still have questions - they're useful for helping me to understand this code better too!